release/release_utils.py

```diff
@@ -390,21 +399,26 @@ def hg_push(repo_root):
     else:
         execute('hg -R %s push' % repo_root)
 
-# Pull the latest changes and update
-def update_project(path):
-    if git_bare_repo_exists_at_path(path):
-        execute('git -C %s fetch' % path)
-    elif is_git(path):
-        execute('git -C %s pull' % path)
+# Clean repo, pull the latest changes and update
```

Comment:
Of course, the merge portion of the update step could fail if the repository is unclean, so the script might not get to the verification step.